nibabel/freesurfer/mghformat.py

```diff
@@ -330,13 +330,10 @@ def set_data_shape(self, shape):
            sequence of integers specifying data array shape
         '''
         dims = self._header_data['dims']
-        # If len(dims) is 3, add a dimension. MGH header always
-        # needs 4 dimensions.
-        if len(shape) == 3:
-            shape = list(shape)
-            shape.append(1)
-            shape = tuple(shape)
-        dims[:] = shape
+        if len(shape) > 4:
+            raise ValueError("Cannot set data shape with more than 4 "
+                             "dimensions")
+        dims[:] = tuple(shape) + (1,) * (4 - len(shape))
```

Comment:
There was previously not a test that tried to create an `MGHImage` with 5 dimensions. `dims[:] = shape` would throw an error in recent versions of numpy, but not the minimum 1.7.1. This change upgrades any <4 dimension objects to 4D, though if you'd prefer I could revert to 3D or 4D only.